cmd/lncli/commands.go

```diff
@@ -495,32 +489,35 @@ func openChannel(ctx *cli.Context) error {
 		RemoteCsvDelay: uint32(ctx.Uint64("remote_csv_delay")),
 	}
 
+	var connectVal string
+	var nodePubHex []byte
+
 	switch {
 	case ctx.IsSet("node_key"):
-		nodePubHex, err := hex.DecodeString(ctx.String("node_key"))
-		if err != nil {
-			return fmt.Errorf("unable to decode node public key: %v", err)
-		}
-		req.NodePubkey = nodePubHex
+		nodePubHex, connectVal, err = decodeNodePubkeyOrURI(
+			ctx.String("node_key"))
 
 	case args.Present():
-		nodePubHex, err := hex.DecodeString(args.First())
-		if err != nil {
-			return fmt.Errorf("unable to decode node public key: %v", err)
-		}
+		nodePubHex, connectVal, err = decodeNodePubkeyOrURI(
```

Comment:
nit: I believe `req.NodePubkey` can be used here directly.

Reply:
I don't think so given that the first arg (like `node_key` above) can either be a pubkey or a URI, and in the latter case we need to unpack it, but maybe I'm missing something.

Reply:
to clarify, what I meant was the following 😄 Not sure if it works?

```go
	switch {
	case ctx.IsSet("node_key"):
		req.NodePubkey, connectVal, err = decodeNodePubkeyOrURI(
			ctx.String("node_key"))

	case args.Present():
		req.NodePubkey, connectVal, err = decodeNodePubkeyOrURI(
			args.First())
		args = args.Tail()
	default:
		return fmt.Errorf("node id argument missing")
	}
```